sdk/core/core/inc/az_json_value.h

```diff
@@ -18,6 +18,9 @@ typedef enum {
   AZ_JSON_VALUE_STRING = 4,
   AZ_JSON_VALUE_OBJECT = 5,
   AZ_JSON_VALUE_ARRAY = 6,
+  // A special case for non-JSON strings. This field is used to serialize `az_span` into JSON
+  // string. Currently, our JSON parser doesn't return this kind of values.
```

Comment:
Should we call it 'non-encoded/non-escaped string'?

Reply:
In this case, I would prefer to rename `AZ_JSON_VALUE_STRING` as well.